### PR TITLE
772 add  is true and is false

### PR DESCRIPTION
**Pull request by timgl**
## Changes
~Needs #760 to be merged first.~
#760 is merged in this PR
- Filtering by Person properties
- Correctly allow filtering of true/false properties
- Allow filtering whether a property is set or not
- Filtering of nested objects and arrays (if slightly crudely by converting to text)

## Checklist
- [x] All querysets/queries filter by Team (if applicable)
- [x] Backend tests (if applicable)

### CODE REVIEW

File: frontend/src/lib/components/PropertyFilters/PropertyValue.js [hunk 1]
```diff
@@ -1,66 +1,65 @@
...
-                )
+
+        this.setState({ input: value, optionsCache: { ...this.state.optionsCache, [value]: 'loading' } })
+        api.get('api/' + this.props.type + '/values/?key=' + key + (value ? '&value=' + value : '')).then(propValues =>
+            this.setState({
+                options: [...new Set([...this.state.options, ...propValues.map(option => option.name)])],
+                optionsCache: { ...this.state.optionsCache, [value]: true },
+            })
         )
     }
```
Comment (EDsCODE) [inline frontend/src/lib/components/PropertyFilters/PropertyValue.js:31]: componentWillReceiveProps can solve the missing values issue here

File: frontend/src/lib/components/PropertyFilters/PropertyValue.js [hunk 2]
```diff
@@ -1,66 +1,65 @@
...
-        this.state = { input: props.value }
+        this.state = {
+            input: '',
+            optionsCache: [],
+            options: props.operator === 'is_set' ? ['true', 'false'] : [],
+        }
+
         this.loadPropertyValues = debounce(this.loadPropertyValues.bind(this), 250)
-        this.ref = React.createRef()
+        if (this.props.operator !== 'is_set') {
```
Comment (EDsCODE) [inline frontend/src/lib/components/PropertyFilters/PropertyValue.js:17]: This is only checking during initialization, but the operator can change while the component is still mounted and the options won't be refreshed

**Review by EDsCODE — CHANGES_REQUESTED** [review]
Person filtering works well. Few thoughts on reviewing/QA'ing is set

3 findings: 

1. There's a NIT with the symbol below.
2. Empty value that can be filtered out
Screen Shot 2020-05-19 at 11 31 56 AM
3. Bug where the true/false values don't change if you try using different filter
Screen Shot 2020-05-19 at 11 36 48 AM

- Add a filter that would expect boolean values (ex: `is_demo`). 
- Apply is set as the filter operator.
- Change the filter property to something else and change the oeprator type away from isset
- The value options will still be true/false

I noted below where the fix can happen

File: frontend/src/lib/utils.js [hunk 3]
```diff
@@ -191,6 +191,7 @@ export const operatorMap = {
     not_icontains: "∌ doesn't contain",
     gt: '> greater than',
     lt: '< lower than',
+    is_set: '! is set',
```
[the remark this answers is not in the document]
Comment (timgl) [reply frontend/src/lib/utils.js:194]: Yeah much prefer that

**Comment by timgl** [comment]
@EDsCODE Fixed all comments, and also fixed cohorts and funnels (both were broken with person properties)

**Review by EDsCODE — APPROVED** [review]

**Pull request merged**